src/components/nav-header-panels/user-menu-panel.component.tsx

```diff
@@ -31,7 +54,22 @@ const UserMenuPanel: React.FC<UserMenuPanelProps> = ({
         allowedLocales={allowedLocales}
         open={expanded}
       ></UserMenu>
-      <Switcher aria-label="Switcher Container">
+      <Switcher className={styles.switcher} aria-label="Switcher Container">
+        <UserAvatarFilledAlt20 />
+        <p>{user.person.display}</p>
+      </Switcher>
+      <Switcher className={styles.switcher} aria-label="Switcher Container">
```

Comment:
This link (`Switcher`) incl. all related logic should be provided in form of an extension component. The extension should come from `esm-login`.

Reason: `esm-login` is currently the host for the location picker. If `esm-login` is gone the location picker is gone. If `esm-login` has the picker on a different address, the current implementation won't work either. Only `esm-login` can be responsible for placing a link to the picker in the user profile.

It's also a good time to introduce the desired flexibility here.